drivers/clock_control/clock_stm32_ll_h7.c

```diff
@@ -298,12 +298,28 @@ static int32_t optimize_regulator_voltage_scale(uint32_t sysclk_freq)
 	defined(CONFIG_CLOCK_STM32_PLL_SRC_HSI) || \
 	defined(CONFIG_CLOCK_STM32_PLL_SRC_CSI)
 
-static int32_t get_vco_output_range(uint32_t vco_input_range)
+static int get_vco_input_range(uint32_t m_div, uint32_t *range)
+{
+	uint32_t vco_freq;
+	vco_freq = PLLSRC_FREQ / m_div;
+	if (1000000UL <= vco_freq && vco_freq <= 2000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_1_2;
+	} else if (2000000UL < vco_freq && vco_freq <= 4000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_2_4;
+	} else if (4000000UL < vco_freq && vco_freq <= 8000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_4_8;
+	} else if (8000000UL < vco_freq && vco_freq <= 16000000UL) {
+		*range = LL_RCC_PLLINPUTRANGE_8_16;
+	} else {
+		return -ERANGE;
+	}
+	return 0;
+}
```

Comment:
I do not see the need of `get_vco_input_range()` function definition, since the calculations are done with preprocessor on the lines  69-81 of this file.
New line is missed also.

Reply:
@Nukersson It is needed when using multiple PLLs.

Reply:
Okay, then another proposal. Please put defines in the top of the file. This defines let programmer read code easier, cause no zeros counting is needed.

```suggestion
#define FREQ_1MHz     1000000UL
#define FREQ_2MHz      2000000UL
#define FREQ_4MHz      4000000UL
#define FREQ_8MHz      8000000UL
#define FREQ_16MHz    16000000UL

static int get_vco_input_range(uint32_t m_div, uint32_t *range)
{
	uint32_t vco_freq;
	vco_freq = PLLSRC_FREQ / m_div;
	if (FREQ_1MHz <= vco_freq && vco_freq <= FREQ_2MHz) {
		*range = LL_RCC_PLLINPUTRANGE_1_2;
	} else if (FREQ_2MHz < vco_freq && vco_freq <= FREQ_4MHz) {
		*range = LL_RCC_PLLINPUTRANGE_2_4;
	} else if (4000000UL < vco_freq && vco_freq <= FREQ_8MHz) {
		*range = LL_RCC_PLLINPUTRANGE_4_8;
	} else if (8000000UL < vco_freq && vco_freq <= FREQ_16MHz) {
		*range = LL_RCC_PLLINPUTRANGE_8_16;
	} else {
		return -ERANGE;
	}
	return 0;
}

```

Reply:
Would it not make more sense to have one define like FREQ_MHZ(x) (x*1000000UL)?

Reply:
> Would it not make more sense to have one define like FREQ_MHZ(x) (x*1000000UL)?

sure

Reply:
Done